### PR TITLE
doc: add response header doc into api

### PR DESCRIPTION
**Pull request by xiang90**
/cc @heyitsanthony @joshix

### CODE REVIEW

File: Documentation/api_v3.md [hunk 1]
```diff
@@ -2,6 +2,30 @@
 
 TODO: finish API doc
 
+## Response Header
+
+All Responses from etcd API has a [response header][response_header] attached. Response header includes the metadata of the response.
```
Comment (joshix) [inline Documentation/api_v3.md]: s/has/have/
s/Response header includes/The response header includes/

File: Documentation/api_v3.md [hunk 2]
```diff
@@ -2,6 +2,30 @@
...
+  uint64 raft_term = 4;
+}
+```
+
+* Cluster_ID - the ID of the cluster that generates the response
+* Member_ID - the ID of the member that generates the response
+* Revision - the revision of the key-value store when the response is generated
+* Raft_Term - the Raft term of the member when the response is generated
+
+Applications can utilize `Cluster_ID` (`Member_ID`) field to ensure always talking to the same cluster (member).
```
Comment (heyitsanthony) [inline Documentation/api_v3.md]: An application may utilize the `Cluster_ID` (`Member_ID`) field to ensure it is communicating with the intended cluster (member).

Comment (joshix) [inline Documentation/api_v3.md]: s/utilize/use/  (or s/utilize/read/)
s/always/they are always/

File: Documentation/api_v3.md [hunk 3]
```diff
@@ -2,6 +2,30 @@
...
+```
+
+* Cluster_ID - the ID of the cluster that generates the response
+* Member_ID - the ID of the member that generates the response
+* Revision - the revision of the key-value store when the response is generated
+* Raft_Term - the Raft term of the member when the response is generated
+
+Applications can utilize `Cluster_ID` (`Member_ID`) field to ensure always talking to the same cluster (member). 
+
+Applications can use the `Revision` to know the latest revision of the key-value store. This is especially useful when applications specifiy a historical revision to do `time travel` query and still wants to know the latest revision from the same response.
```
Comment (joshix) [inline Documentation/api_v3.md]: s/specifiy/specify/
s/to do/to do a/
s/wants/want/
s/from/of/

File: Documentation/api_v3.md [hunk 4]
```diff
@@ -2,6 +2,30 @@
...
+
+Applications can use `Raft_Term` to detect any newly leader election happened in the cluster.
```
Comment (joshix) [inline Documentation/api_v3.md]: "Applications can use the `Raft_Term` of the metadata to detect if a new leader election has occurred."

**Comment by joshix** [comment]
Few nits, inline

File: Documentation/api_v3.md (hunk 4 above)
Comment (heyitsanthony) [inline Documentation/api_v3.md]: to detect when the cluster completes a new leader election.

**Comment by xiang90** [comment]
@joshix @heyitsanthony All fixed. Merging.

**Pull request merged**